src/devices/Arduino/ArduinoBoard.cs

```diff
@@ -0,0 +1,413 @@
+            }
+
+            if (!SupportedPinConfigurations.Any(x => x.PinModes.Contains(SupportedMode.SPI)))
+            {
+                throw new NotSupportedException("No Pins with SPI support found. Is the SPI module loaded?");
+            }
+
+            return new ArduinoSpiDevice(this, settings);
+        }
+
+        /// <summary>
+        /// Creates a PWM channel.
+        /// </summary>
+        /// <param name="chip">Must be 0.</param>
+        /// <param name="channel">Pin number to use</param>
+        /// <param name="frequency">This value is ignored</param>
+        /// <param name="dutyCyclePercentage">The duty cycle as a fraction.</param>
+        /// <returns></returns>
+        public virtual PwmChannel CreatePwmChannel(
+            int chip,
+            int channel,
+            int frequency = 400,
+            double dutyCyclePercentage = 0.5)
+        {
+            return new ArduinoPwmChannel(this, chip, channel, frequency, dutyCyclePercentage);
+        }
+
+        /// <summary>
+        /// Creates an anlog controller for this board.
+        /// </summary>
+        /// <param name="chip">Must be 0</param>
+        /// <returns>An <see cref="AnalogController"/> instance</returns>
+        public virtual AnalogController CreateAnalogController(int chip)
+        {
+            return new ArduinoAnalogController(this, SupportedPinConfigurations, PinNumberingScheme.Logical);
+        }
+
+        /// <summary>
+        /// Special function to read DHT sensor, if supported
+        /// </summary>
+        /// <param name="pinNumber">Pin Number</param>
+        /// <param name="dhtType">Type of DHT Sensor: 11 = DHT11, 22 = DHT22, etc.</param>
+        /// <param name="temperature">Temperature</param>
+        /// <param name="humidity">Relative humidity</param>
+        /// <returns>True on success, false otherwise</returns>
+        public bool TryReadDht(int pinNumber, int dhtType, out Temperature temperature, out RelativeHumidity humidity)
+        {
+            if (!_supportedPinConfigurations[pinNumber].PinModes.Contains(SupportedMode.DHT))
+            {
+                temperature = default;
+                humidity = default;
+                return false;
+            }
+
+            return Firmata.TryReadDht(pinNumber, dhtType, out temperature, out humidity);
+        }
+
+        /// <summary>
+        /// Standard dispose pattern
+        /// </summary>
+        protected virtual void Dispose(bool disposing)
+        {
+            // Do this first, to force any blocking read operations to end
+            if (_dataStream != null)
+            {
+                _dataStream.Dispose();
+                _dataStream = null!;
+            }
+
+            if (_serialPort != null)
+            {
+                _serialPort.Dispose();
+                _serialPort = null;
+            }
+
+            if (_firmata != null)
+            {
+                _firmata.OnError -= FirmataOnError;
```

Comment:
NIT: you can just make sure that on the Firmata class you clear out the OnError handler when calling Dispose instead of doing it here.